src/TheMovie/TheMovie/ViewModels/UpcomingMovieViewModel.cs

```diff
@@ -0,0 +1,198 @@
+        public DelegateCommand<UpcomingMovieModel> ItemTappedCommand { get; set; }
+
+        #endregion
+
+        #region Services
+
+        private readonly IMovieService _movieService;
+        private readonly IGenreService _genreService;
+
+        #endregion
+
+        public UpcomingMovieViewModel(INavigationService navigationService, IPageDialogService pageDialogService, IMovieService movieService, IGenreService genreService)
+            : base(navigationService, pageDialogService)
+        {
+            _movieService = movieService;
+            _genreService = genreService;
+            Title = "Upcoming Movies";
+            SearchPlaceHolder = "Search your movie :)";
+            UpcomingMovies = new ObservableCollection<UpcomingMovieModel>();
+            SearchCommand = new Command(async () => await ExecuteSearchCommand());
+            UpcomingMovieRefreshCommand = new Command(async () => await ExecuteUpcomingMovieRefreshCommand());
+            ItemAppearingCommand = new DelegateCommand<UpcomingMovieModel>(ExecuteItemAppearingCommand);
+            ItemTappedCommand = new DelegateCommand<UpcomingMovieModel>(ExecuteItemTappedCommand);
+            InitializeUpcomingMovies();
+        }
+
+        #region Methods 
+
+        private async Task GetUpcomingMovies(int pageMovie = 1)
+        {
+            var movies = await _movieService.GetAll(pageMovie);
+
+            if (movies.UpcomingMovies.Count() > 0)
+            {
+                _pageUpcomingMovies = movies.Page;
+                _totalPagesUpcomingMovies = movies.TotalPages;
+
+                var genres = await _genreService.GetAll();
+
+                foreach (var movie in movies.UpcomingMovies)
+                {
+                    foreach (var genreId in movie.GenreIds)
+                    {
+                        movie.Genres += $"{genres.GenreMovies.FirstOrDefault(x => x.Id == genreId).Name}\n";
+                    }
+
+                    UpcomingMovies.Add(movie);
+                }
+            }
+        }
+
+        private async void InitializeUpcomingMovies()
+        {
+            IsLoading = true;
+
+            UpcomingMovies.Clear();
+
+            await GetUpcomingMovies();
+
+            IsLoading = false;
+        }
+
+        #endregion
+
+        #region Methods Commands
+
+        private async Task ExecuteSearchCommand()
+        {
+            _isSearchUpcomingMovies = true;
+
+            var searchResult = UpcomingMovies.Where(x => x.Title.ToUpper().Contains(SearchText.ToUpper())).ToList();
+
+            if (searchResult.Count > 0)
+            {
+                UpcomingMovies.Clear();
+
+                foreach (var search in searchResult)
+                {
+                    UpcomingMovies.Add(search);
+                }
+            }
+            else
+            {
+                await PageDialogService.DisplayAlertAsync("Ooops!", "Movie not found :(", "Ok");
```

Comment:
Could you explain the difference between these two lines and if I made this method `private Task` and replace these lines with:
```
return PageDialogService.DisplayAlertAsync ...
```

Reply:
I'm not sure, but, would you like to do a private method with "return PageDialogService.DisplayAlertAsync"?

The method would have to be asynchronous, because the return of the "PageDialogService.DisplayAlertAsync" method is asynchronous.